### PR TITLE
Fix for Issue #1414: DSSTATS reports incorrectly that a data source does not exist

### PR DESCRIPTION
**Pull request by netniV**
This fix corrects the issue with DSSTATS spamming logs claiming that the file does not exist when it does in fact exist.  This is due to the incorrect usage of comparing a string to an integer.

Also added a statscache clear just to make sure the latest state is retrieved for a file at the time of processing.

### CODE REVIEW

File: lib/dsstats.php [hunk 1]
```diff
@@ -156,11 +156,12 @@ function for storage into the respective database table.
 function dsstats_obtain_data_source_avgpeak_values($rrdfile, $interval, $rrdtool_pipe) {
 	global $config;
 
-	$use_proxy = (read_config_option('storage_location') != '' ? true:false);
+	$use_proxy = (read_config_option('storage_location') ? true:false);
 
 	if ($use_proxy) {
 		$file_exists = rrdtool_execute("file_exists $rrdfile", true, RRDTOOL_OUTPUT_BOOLEAN, $rrdtool_pipe, 'DSSTATS');
 	}else {
```
Comment (pautiina) [inline lib/dsstats.php]: Maybe need add space
`} else {`

Comment (netniV) [reply lib/dsstats.php]: Yeah the function should probably have a format update but I haven't changed that part 👍

**Comment by pautiina** [comment]
https://github.com/Cacti/cacti/commit/a32fe7b0404bda830e9f014a319a827eb131cdcd#r28318090

**Comment by netniV** [comment]
Can you open that as a new issue so we can make the associated fix against it.

**Pull request merged**